sdk/state/update.go

```diff
@@ -126,7 +121,7 @@ func (c *Channel) ConfirmPayment(p *Payment) (*Payment, error) {
 
 	p.CloseSignatures = append(p.CloseSignatures, txClose.Signatures()...)
 	p.DeclarationSignatures = append(p.DeclarationSignatures, txDecl.Signatures()...)
-	c.amount.Amount = newBalance
+	c.lastConfirmedPayment = p
```

Comment:
❗ This records the payment as confirmed, but it is not confirmed until all the signatures are on the payment. I think we need a check before these few lines to check that if the remote has not signed the declaration, it should return an error. Similar to this code in the open func:

https://github.com/stellar/experimental-payment-channels/blob/2d90876b9f238302e57d0b3a2927e1a7e9c4d4e3/sdk/state/state_open.go#L142-L146

Reply:
That's a good point. To avoid merge conflicts I can leave this change as part of the returning `payment, fullySigned, err :=` ticket here: https://github.com/stellar/experimental-payment-channels/issues/65

lmk if that sounds good, otherwise I can change it here first

Reply:
That's good with me.